src/components/common/UserProfileEdit/UserMainSettings.tsx

```diff
@@ -94,127 +73,99 @@ const validationSchema = object({
 
 type FormValues = InferType<typeof validationSchema>;
 
+const handleSubmit = (
+  updatedProfile: FormValues,
+  walletAddress: string,
+  editUser: UpdateUserProfileMutationFn,
+) => {
+  editUser({
+    variables: {
+      input: {
+        id: walletAddress,
+        ...updatedProfile,
+        email: updatedProfile.email || null,
+      },
+    },
+  });
+};
+
 const UserMainSettings = ({
   user: { walletAddress, profile },
   user,
 }: Props) => {
-  const { updateUser } = useAppContext();
-
-  const userProfile = excludeTypenameKey(profile);
   const [showSnackbar, setShowSnackbar] = useState<boolean>(false);
-
-  const [editUser, { error, loading, called }] = useUpdateUserProfileMutation();
-  const defaultValues = {
-    email: profile?.email || '',
-    displayName: profile?.displayName || '',
-    bio: profile?.bio || '',
-    website: profile?.website || '',
-    location: profile?.location || '',
-  };
-
-  const handleSubmit = (updatedProfile: FormValues) => {
-    const valuesToSubmit = {
-      email: updatedProfile.email === '' ? null : updatedProfile.email,
-    };
-
-    editUser({
-      variables: {
-        input: {
-          id: walletAddress,
-          ...userProfile,
-          ...updatedProfile,
```

Comment:
Here we can just spread the updatedProfile, we don't need to also spread the previous user profile.